src/teams.rs

```diff
@@ -4,84 +4,232 @@ use std::collections::BTreeMap;
 
 use diesel::prelude::*;
 use toml;
+use serde::de;
 
 use super::DB_POOL;
 use domain::github::GitHubUser;
 use error::*;
 
+//==============================================================================
+// Public API
+//==============================================================================
+
 lazy_static! {
-    pub static ref TEAMS: Teams = {
-        let teams_file =
-            include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/teams.toml"));
-        let teams_from_file: TeamsFromFile =
-            toml::from_str(teams_file).expect("couldn't parse teams");
-
-        let mut teams = BTreeMap::new();
-
-        for (label, team_from_file) in teams_from_file {
-            let label = TeamLabel(label);
-            let team = team_from_file.validate()
-                .expect("unable to verify team member from database.
-if you're running this for tests, make sure you've pulled github users from prod");
-            teams.insert(label, team);
-        }
+    pub static ref SETUP: RfcbotConfig = read_rfcbot_cfg_validated();
+}
+
+#[derive(Debug, Deserialize)]
+pub struct RfcbotConfig {
+    fcp_behaviors: BTreeMap<String, FcpBehavior>,
+    teams: BTreeMap<TeamLabel, Team>,
+}
+
+impl RfcbotConfig {
+    /// Retrive an iterator over all the team labels.
+    pub fn team_labels(&self) -> impl Iterator<Item = &TeamLabel> {
+        self.teams.keys()
+    }
+
+    /// Retrive an iterator over all the (team label, team) pairs.
+    pub fn teams(&self) -> impl Iterator<Item = (&TeamLabel, &Team)> {
+        self.teams.iter()
+    }
+
+    /// Are we allowed to auto-close issues after F-FCP in this repo?
+    pub fn ffcp_auto_close(&self, repo: &str) -> bool {
+        self.fcp_behaviors.get(repo).map(|fcp| fcp.close).unwrap_or_default()
+    }
 
-        teams
-    };
+    /// Are we allowed to auto-postpone issues after F-FCP in this repo?
+    pub fn ffcp_auto_postpone(&self, repo: &str) -> bool {
+        self.fcp_behaviors.get(repo).map(|fcp| fcp.postpone).unwrap_or_default()
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub struct FcpBehavior {
+    #[serde(default)]
+    pub close: bool,
+    #[serde(default)]
+    pub postpone: bool,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct Team {
+    // FIXME: The two following first fields don't seem to be used anywhere...
+    // If this is intended, document why.
+    name: String,
+    ping: String,
```

Comment:
I think this code might predate `pub(crate)` and so I didn't notice when those fields stopped being used :).

Feel free to remove them, I don't think this struct is used in API responses.

Reply:
Commented them out for now so that they are not wasting (a small bit of) RAM :)
Keeping them in `rfcbot.toml` in case we find future uses for them (maybe in better public messages, etc...)